zerver/lib/push_notifications.py

```diff
@@ -662,15 +664,18 @@ def send_notifications_to_bouncer(
         # The server may have updated our understanding of whether
         # push notifications will work.
         assert isinstance(remote_realm_dict, dict)
+        can_push = remote_realm_dict["can_push"]
         do_set_realm_property(
             user_profile.realm,
             "push_notifications_enabled",
-            remote_realm_dict["can_push"],
+            can_push,
             acting_user=None,
         )
         do_set_push_notifications_enabled_end_timestamp(
             user_profile.realm, remote_realm_dict["expected_end_timestamp"], acting_user=None
         )
+        if can_push:
+            record_push_notifications_recently_working()
```

Comment:
I thought about whether we should also do some remembering like that in the `send_server_data_to_push_bouncer` codepath, but it probably doesn't make sense, even if only because it only runs once an hour anyway?

Reply:
Yeah, that's probably not necessary.